### PR TITLE
Quiz `.json` files as database for tests

### PR DESCRIPTION
**Pull request by YacineSteeve**
Each `quiz_subject.json` represents an array of 5 questions related to `subject`:

Here is an example of question in `quiz_linux.json`:

```json
[
  {
       "id": 663,
        "question": "What is the output of the command \u2018umask \u2013S\u2019?",
        "description": null,
        "answers": {
            "answer_a": "Shows mask value using symbolic notion.",
            "answer_b": "Shows mask value using octal values.",
            "answer_c": "Removes the current mask value.",
            "answer_d": "Sets new mask value.",
            "answer_e": null,
            "answer_f": null
        },
        "multiple_correct_answers": "false",
        "correct_answers": {
            "answer_a_correct": "true",
            "answer_b_correct": "false",
            "answer_c_correct": "false",
            "answer_d_correct": "false",
            "answer_e_correct": "false",
            "answer_f_correct": "false"
        },
        "correct_answer": "answer_a",
        "explanation": null,
        "tip": null,
        "tags": [
            {
                "name": "BASH"
            }
        ],
        "category": "Linux",
        "difficulty": "Easy"
  }
  ...
]
```

### CODE REVIEW

File: backend/api/temp/quiz_bash.json [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        "id": 9,
+        "question": "#!\/bin\/bash is commonly called as",
+        "description": null,
+        "answers": {
+            "answer_a": "shebang",
+            "answer_b": "hashbang",
+            "answer_c": "Script Initialiser",
+            "answer_d": "None of the Above",
+            "answer_e": null,
```
Comment (chrichri17) [inline backend/api/temp/quiz_bash.json]: Do we really need to make them `null`? Let's just remove those fields. inexistence should imply `null`

File: backend/api/temp/quiz_bash.json [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        "tip": null,
+        "tags": [
+            {
+                "name": "BASH"
+            }
+        ],
+        "category": "bash",
+        "difficulty": "Easy"
+    }
+]
```
Comment (chrichri17) [inline backend/api/temp/quiz_bash.json]: nit: end of line

There should be a blank line at the end of the file

File: backend/api/temp/quiz_cms.json [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            {
+                "name": "WordPress"
+            }
+        ],
+        "category": "CMS",
+        "difficulty": "Easy"
+    },
+    {
+        "id": 261,
+        "question": "What should one use for plugin development \u2014 custom post types or custom database tables?",
```
Comment (chrichri17) [inline backend/api/temp/quiz_cms.json]: nit: unicode character `\u2014`

File: backend/api/temp/quiz_cms.json [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        "description": null,
+        "answers": {
+            "answer_a": "custom post type",
+            "answer_b": "custom database tables",
+            "answer_c": "There is no specific preference for plugin development; it depends on what type of plugin\u2019s one has to develop.",
```
Comment (chrichri17) [inline backend/api/temp/quiz_cms.json]: nit unicode character

**Review by chrichri17 — CHANGES_REQUESTED** [review]
Looks good but:
- rename the directory with something more explicit like `data` or `seed_data`
- we should maybe rethink the model. Let me wrap my head around it and see if we can go with the current one

File: backend/api/temp/quiz_bash.json (hunk 1 above)
Comment (YacineSteeve) [reply backend/api/temp/quiz_bash.json]: You're right.
I haven't made any changes to the format or values yet, they are still in the state I took fetched them.

**Review by chrichri17 — COMMENTED** [review]
Looks good. But we should probably design our own interface and then generate examples

**Comment by chrichri17** [comment]
This is a start

```typescript

interface Question {
    /** Id of the question */
    id: string;
    /** What kind of question */
    kind: QuestionKind;
    /** Spec of the question: specify what are the attributes of the question. It depends on the question kind ! */
    spec: QuestionSpec;
    /** Object that specifies how to grade for this question */
    grading: Grading;
    /** Is my question easy, medium or hard ? */
    difficulty: Difficulty;
    /** List of optional tags associated with the question */
    tags: string[]; // not really needed in the MVP
    /** Hint given to help answering the question */
    hint?: string;
}

type Grading = {
    /** How much a correct answer count for in the total score */
    point_value: number;
    /** Correct answers of the question */
    correct_answers: Answer[];
    /** Feedback after the question is answered. It can be positive, negative or general. */
    feedback: Feedback;
}

type Feedback = {
    /** Feedback when answer is wrong */
    when_wrong?: string;
    /** Feedback when answer is correct */
    when_right?: string;
    /** General feedback displayed whether the answer is wrong or correct. */
    general: string;
}

/** To Be Determined: I'm still not sure about this */
type Answer = {
    value: string;
}

/** Different kind of question wue might have */
type QuestionKind = "choice_question" | "text_question" | "upload_question" | "code_question";

/** Question difficulty: easy, medium or hard */
type Difficulty = "Easy" | "Medium" | "Hard";

/** And here comes the actual work: design each question spec, i.e how to represent each question */
type QuestionSpec = ChoiceQuestionSpec | TextQuestionSpec | UploadQuestionSpec | CodeQuestionSpec;

type ChoiceQuestionSpec = null;

type TextQuestionSpec = null;

type UploadQuestionSpec = null;

type CodeQuestionSpec = null;
```

**Pull request merged**